detekt-rules-complexity/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/NestedScopeFunctions.kt

```diff
@@ -65,7 +66,9 @@ class NestedScopeFunctions(config: Config = Config.empty) : Rule(config) {
     }
 
     override fun visitNamedFunction(function: KtNamedFunction) {
-        function.accept(FunctionDepthVisitor())
+        if (bindingContext != BindingContext.EMPTY) {
```

Comment:
```suggestion
        if (bindingContext == BindingContext.EMPTY) return
```

Let's keep the same style as all the other rules 👍